File: AccountTransferMvpApp/AccountTransferMvpApp/Presenters/AccountTransferPresenter.m
#import "AccountTransferPresenter.h"
#import "ILocalAccountRepository.h"
#import "IRemoteAccountRepository.h"
#import "IAccountTransferView.h"


@implementation AccountTransferPresenter {

    id <IRemoteAccountRepository> _remoteAccountRepository;
    id <ILocalAccountRepository> _localAccountRepository;
}
@synthesize accountTransferView = _accountTransferView;


- (id)initWithRemote:(id <IRemoteAccountRepository>)remoteAccountRepository andLocal:(id <ILocalAccountRepository>)localAccountRepository {
    if (self = [super init]) {
        _remoteAccountRepository = remoteAccountRepository;
        _localAccountRepository = localAccountRepository;
    }
    return self;
}

- (void)transferAmount {
    NSNumber *amount = [_accountTransferView getTransferAmount];
    [_remoteAccountRepository transferFromAmount:amount];
    [_localAccountRepository transferToAmount:amount];
    [_accountTransferView setDisplayMessage:[NSString stringWithFormat:@"$%@ transferred.", amount]];
}
@end

File: AccountTransferMvpApp/App/src/com/gaddzeit/presenters/AccountTransferPresenter.java
package com.gaddzeit.presenters;

import com.gaddzeit.repositories.ILocalAccountRepository;
import com.gaddzeit.repositories.IRemoteAccountRepository;
import com.gaddzeit.views.IAccountTransferView;

public class AccountTransferPresenter {
    private final IRemoteAccountRepository _remoteAccountRepository;
    private final ILocalAccountRepository _localAccountRepository;
    private IAccountTransferView _accountTransferView;

    public AccountTransferPresenter(IRemoteAccountRepository remoteAccountRepository, ILocalAccountRepository localAccountRepository) {
        _remoteAccountRepository = remoteAccountRepository;
        _localAccountRepository = localAccountRepository;
    }

    public void setView(IAccountTransferView accountTransferView) {

        _accountTransferView = accountTransferView;
    }

    public void transferAmount() {
        final double amount = _accountTransferView.getTransferAmount();
        _remoteAccountRepository.withdrawAmount(amount);
        _localAccountRepository.depositAmount(amount);
        _accountTransferView.setDisplayMessage(String.format("$%s transferred.", amount));
    }
}
